stdlib/public/core/Sequence.swift

```diff
@@ -800,6 +800,18 @@ extension Sequence where Element : Equatable {
   }
 }
 
+extension ContiguousArray {
+  /// Rotates array in place to the left by specified distance.
+  @usableFromInline
+  mutating func _rotate(left distance: Int) {
+    guard distance > 0 else { return }
+    let i = index(startIndex, offsetBy: distance)
```

Comment:
This is not necessarily the best way to rotate, and is probably not the API we want to commit to supporting even internally.  Please try to use what's already written in the algorithms prototype, and especially read [this comment](https://github.com/apple/swift/blob/bf1df4fdf8881d0ff50a552b99c7ffb3afad66b7/test/Prototypes/Algorithms.swift#L193-L203) and test accordingly.

Reply:
I guess it's better than to split this and do just the consistent use of `ContiguousArray` in this PR, and tackle the in-place rotation in a separate PR, right?